surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+# Survey questions
+
+Whether or not you use Rust Programming Language <https://rust-lang.org> today, we want to hear from you!
+
+The Rust Community Team has created this survey to help us gauge how we're doing, what can be improved, and how we can best engage with all of you as we move forward.
+
+This is your chance to have a say in the development priorities of Rust.
+
+Your answers will be anonymous. Any personal data you submit as a part of this survey will be handled in accordance with our policy as described in our [Frequently Asked Questions](https://github.com/rust-lang/surveys/blob/main/documents/Community-Survey-FAQ.md).
+
+We estimate it will take about 10-25 minutes to complete.
+
+## Rust Usage
+
+### Do you use Rust?
+
+Type: select one
+
+- Yes, I use Rust (for any purpose, even if you're just learning) [`NEXT`](#your-rust-experience)
+- No, I don't currently use Rust, but I have in the past [`NEXT`](#for-previous-rust-users)
+- No, I have never used Rust [`NEXT`](#for-non-rust-users)
+
+> **justification**
+>
+> Fundamental for cohort analysis
+
+## For previous Rust users
+
+### Why did you stop using Rust?
+
+Type: select all that apply
+
+- Rust did not help me achieve my goals
+- Missing language features
+- Missing libraries
+- Missing tools
+- Too difficult or complex to become productive writing it
+- Community was rude, unwelcoming, or otherwise off-putting
+- I prefer to use another language
+- I no longer have the opportunity to use Rust due to factors outside of my control
+- Other
+
+### Tell us more:
+
+Type: free text
+
+> **SURVEY FLOW**
+>
+> Skip to `### To what extent is Rust currently being used by your company?`
+
+
+## For non-Rust users
+
+### Why don't you use Rust?
+
+Type: select all that apply
+
+- Rust did not help me achieve my goals
+- Missing language features
+- Missing libraries
+- Missing tools
+- Too difficult to learn or learning will take too much time
+- Community was rude, unwelcoming, or otherwise off-putting
+- I prefer to use another language
+- I can't use Rust due to factors outside of my control
+- I haven't got around to it
+- Other
+
+### Tell us more:
+
+Type: free text
+
+> **SURVEY FLOW**
+>
+> Skip to `### To what extent is Rust currently being used by your company?`
+
+## Your Rust experience
+
+### On average, how often do you use Rust?
+
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> **justification**
+>
+> This can be useful demographic information as a proxy for how "important" Rust is
+> in someone's technical life.
+>
+> We deliberately use calendar time for this question to gauge how "serious" the
+> programmer's use of Rust is. This does mean that we will group together, for example,
```

Comment:
Can we have a separate question for "how often do you program" and then look at the results together?

Reply:
IIUC your suggestion:
- On average, how often do you do coding?
- On average, how often do you do coding _in Rust_?

(maybe improved this wording I pull out on the spot)

Reply:
Yep, that's the suggestion :)

Reply:
This doesn't flow well into the questionnaire, as this section begins with a title "Your Rust experience". But I went through git history and there were some concerns previously about putting together Rust and non-Rust programming time. So adding a new question for it is fine by me, we'd just need to figure out where to put it. Maybe just adding it as a question directly after this one is ok.

Reply:
Filed [TODO](https://github.com/rust-lang/surveys/issues/239) for this. I'm not even sure if we're able to cross-reference people that selected some answer in one question with their answer in another question though.. :)